File: targets/go/fiber/query.go
// query: query string parsing and coercion, isolated from any use of the values.
//
// Fiber's own binder fills the struct: c.Bind().Query reads the `query:` tag through
// gofiber/schema. The struct is both what Fiber fills and what the handler answers, so the
// response is the bound values and nothing copies one shape into another.
//
// SkipValidation, because the StructValidator in the config runs after every bind and these
// two arms have no rules to run: the family measures the parse, and a validator walking
// eight fields with nothing to say would be measured alongside it.
//
// Fiber decides what a value it cannot bind is: a missing parameter leaves the field at its
// zero value, and one that will not parse comes back as a *fiber.BindError. That never
// fires on the endpoint set, whose query strings are always complete and well-formed, but
// it is the framework's line and not this file's.
package main

import (
	"github.com/gofiber/fiber/v3"
)

// rb:wiring query.*
type queryOne struct {
	Page int `query:"page" json:"page"`
}

// rb:wiring query.*
type queryMany struct {
	Page     int    `query:"page"      json:"page"`
	Size     int    `query:"size"      json:"size"`
	Status   string `query:"status"    json:"status"`
	Category string `query:"category"  json:"category"`
	Sort     string `query:"sort"      json:"sort"`
	Q        string `query:"q"         json:"q"`
	MinPrice int    `query:"min_price" json:"min_price"`
	MaxPrice int    `query:"max_price" json:"max_price"`
}

// What domain.filter pages by. Not a response shape, so it carries no json tags.
// rb:wiring domain.*
type orderFilter struct {
	Page   int    `query:"page"`
	Size   int    `query:"size"`
	Status string `query:"status"`
}

// bindQuery fills out from the query string, answering the failure itself if there is one.
// The same 400 shape as a body Fiber could not bind, because it is the same binder refusing.
// rb:wiring query.*,domain.*
func bindQuery(c fiber.Ctx, out any) bool {
	if err := c.Bind().SkipValidation(true).Query(out); err != nil {
		_ = c.Status(400).JSON(fiber.Map{"error": "invalid_query", "detail": err.Error()})
		return false
	}
	return true
}

func registerQuery(app *fiber.App) {
	app.Get("/query/one", func(c fiber.Ctx) error {
		var q queryOne
		if !bindQuery(c, &q) {
			return nil
		}
		return c.JSON(q)
	})

	app.Get("/query/many", func(c fiber.Ctx) error {
		var q queryMany
		if !bindQuery(c, &q) {
			return nil
		}
		return c.JSON(q)
	})
}

File: targets/dotnet/minimal-apis/Routes/Query.cs
using Microsoft.AspNetCore.Mvc;
using RequestBench.Domain;

namespace RequestBench.MinimalApis.Routes;

/// <summary>
/// query: query string parsing and coercion, isolated from any use of the values.
///
/// Minimal APIs bind a query parameter by declaring it on the handler: the name and the
/// type are the binding, and the framework converts what it parsed before the handler runs.
/// [FromQuery(Name = ...)] is only for the two parameters whose name on the wire is not the
/// C# one.
///
/// The parameters are not nullable and carry no default, so the framework decides what a
/// missing or unconvertible one is: its own 400, before the handler. The endpoint set sends
/// neither.
/// </summary>
public static class Query
{
    public static void Map(WebApplication app)
    {
        app.MapGet("/query/one", (int page) => new QueryOne(page));

        app.MapGet("/query/many", (int page, int size, string status, string category,
                                   string sort, string q,
                                   [FromQuery(Name = "min_price")] int minPrice,
                                   [FromQuery(Name = "max_price")] int maxPrice) =>
            new QueryMany(page, size, status, category, sort, q, minPrice, maxPrice));
    }
}
